### PR TITLE
refactor : use generic function for table ExpandableConfig

### PR DESCRIPTION
**Pull request by Sachin-chaurasiya**
### Describe your changes :

I worked on refactoring the expandable table config to use the generic function for table ExpandableConfig.

#
### Type of change :

- [x] Improvement

#
### Frontend Preview (Screenshots) :
image

image

image

image

image

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
@open-metadata/ui

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/TeamDetails/TeamHierarchy.tsx [hunk 1]
```diff
@@ -189,7 +153,10 @@ const TeamHierarchy: FC<TeamHierarchyProps> = ({
           components={TABLE_CONSTANTS}
           data-testid="team-hierarchy-table"
           dataSource={data}
-          expandable={expandableConfig}
+          expandable={{
+            ...getTableExpandableConfig<Team>(),
+            ...expandableConfig,
+          }}
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/TeamDetails/TeamHierarchy.tsx:159]:
```suggestion
          expandable={expandableConfig}
```

And just spread `getTableExpandableConfig` to `expamdableConfig` only

File: openmetadata-ui/src/main/resources/ui/src/utils/TableUtils.tsx [hunk 2]
```diff
@@ -360,20 +361,17 @@ export const getTestResultBadgeIcon = (status?: TestCaseStatus) => {
   }
 };
 
-export function getTableExpandableConfig<
-  T extends { children?: T[] }
->(): ExpandableConfig<T> {
+export function getTableExpandableConfig<T>(): ExpandableConfig<T> {
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/utils/TableUtils.tsx:364]: I don't think we need generic type <T> anymore let's remove it

**Review by chirag-madlani — CHANGES_REQUESTED** [review]

File: openmetadata-ui/src/main/resources/ui/src/components/TeamDetails/TeamHierarchy.tsx [hunk 3]
```diff
@@ -134,13 +134,14 @@ const TeamHierarchy: FC<TeamHierarchyProps> = ({
...
     () => ({
+      ...getTableExpandableConfig<Team>(),
       onExpand: (isOpen, record) => {
         if (isOpen && isEmpty(record.children)) {
           onTeamExpand(false, record.fullyQualifiedName, true);
         }
       },
     }),
-    [onTeamExpand]
+    [onTeamExpand, getTableExpandableConfig]
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/TeamDetails/TeamHierarchy.tsx:144]: `getTableExpandableConfig` is external function so no need to be added here

**Review by chirag-madlani — CHANGES_REQUESTED** [review]

File: openmetadata-ui/src/main/resources/ui/src/components/TeamDetails/TeamHierarchy.tsx (hunk 3 above)
Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/components/TeamDetails/TeamHierarchy.tsx:144]: My bad, I will remove that.

**Review by ShaileshParmar11 — DISMISSED** [review]

**Review by chirag-madlani — APPROVED** [review]

**Pull request merged**